src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeService.java

```diff
@@ -65,7 +76,10 @@ private UUID createEnvelope(String containerName, String blobName, Instant blobC
     }
 
     @Transactional
-    public void markEnvelopeAsDeleted(UUID envelopeId) {
-        envelopeRepository.markAsDeleted(envelopeId);
+    public void markEnvelopeAsDeleted(Envelope envelope, boolean isFromRejected) {
+        envelopeRepository.markAsDeleted(envelope.id);
+
+        var newEvent = isFromRejected ? Event.DELETED_FROM_REJECTED : Event.DELETED;
+        eventRecordRepository.insert(new NewEventRecord(envelope.container, envelope.fileName, newEvent));
```

Comment:
aren't file files deleted from rejected containers already marked as deleted??

Reply:
ah. i'm creating an event of deletion from rejected container separately. this flag is not needed